app/lib/service/security_advisories/backend.dart

```diff
@@ -0,0 +1,113 @@
+// Copyright (c) 2023, the Dart project authors.  Please see the AUTHORS file
+// for details. All rights reserved. Use of this source code is governed by a
+// BSD-style license that can be found in the LICENSE file.
+
+import 'dart:async';
+import 'dart:convert';
+
+import 'package:basics/basics.dart';
+import 'package:gcloud/service_scope.dart' as ss;
+import 'package:logging/logging.dart';
+import 'package:pub_dev/service/security_advisories/models.dart';
+import 'package:pub_dev/shared/datastore.dart';
+import 'package:pub_dev/shared/redis_cache.dart';
+
+final _logger = Logger('security_advisories.backend');
+
+/// Sets the security advisory backend service.
+void registerSecurityAdvisoryBackend(SecurityAdvisoryBackend backend) =>
+    ss.register(#_securityAdvisoryBackend, backend);
+
+/// The active security advisory backend service.
+SecurityAdvisoryBackend get securityAdvisoryBackend =>
+    ss.lookup(#_securityAdvisoryBackend) as SecurityAdvisoryBackend;
+
+class SecurityAdvisoryBackend {
+  final DatastoreDB _db;
+
+  SecurityAdvisoryBackend(this._db);
+
+  Future<List<OSV>> lookupSecurityAdvisories(
+    String package,
+  ) async {
+    return (await cache.securityAdvisories(package).get(() async {
+      final query = _db.query<SecurityAdvisory>()
+        ..filter('affectedPackages =', package);
+      return query.run().map((e) => e.osvJsonBlob!).toList();
+    }))!;
+  }
+
+  Future<SecurityAdvisory?> lookupById(String id) async {
+    final key = _db.emptyKey.append(SecurityAdvisory, id: id);
+    return _db.lookupOrNull<SecurityAdvisory>(key);
+  }
+
+  List<String> validateAdvisoryErrors(OSV osv) {
```

Comment:
Can't we do unit tests for this function.

We could even make this a helper function in a separate file, you don't need to setup the server to do this at all.

Reply:
Added todo.